src/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.tsx

```diff
@@ -0,0 +1,194 @@
+import Checkbox from '@mui/material/Checkbox'
+import Box from '@mui/material/Box'
+import { FC, SyntheticEvent } from 'react'
+import { useTranslation } from 'react-i18next'
+import AppButton from '~/components/app-button/AppButton'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { styles } from '~/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.styles'
+import {
+  ButtonVariantEnum,
+  ComponentEnum,
+  ProfessionalCategory,
+  ProfessionalSubject,
+  ProficiencyLevelEnum
+} from '~/types'
+import useForm from '~/hooks/use-form'
+import AppAutoComplete from '~/components/app-auto-complete/AppAutoComplete'
+import AddIcon from '@mui/icons-material/Add'
+import { professionalSubjectTemplate } from '~/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.constants'
+import ProficiencyLevelSelect from '~/containers/proficiency-level-select/ProficiencyLevelSelect'
+
+interface ValueWithHandler<T> {
+  value: T
+  handleChange: (value: T) => void
+}
+
+interface SubjectGroupProps {
+  isCategoryDisabled: boolean
+  subjectName: ValueWithHandler<string | null>
+  proficiencyLevels: ValueWithHandler<ProficiencyLevelEnum[]>
+}
+
+function SubjectGroup({
+  isCategoryDisabled,
+  subjectName,
+  proficiencyLevels
+}: SubjectGroupProps) {
+  const { t } = useTranslation()
+
+  return (
+    <Box sx={styles.item}>
+      <Box sx={styles.checkboxGroup}>
+        <Checkbox checked disabled={isCategoryDisabled} sx={styles.checkbox} />
+        <AppAutoComplete
+          fullWidth
+          onChange={(_, value) => subjectName.handleChange(value)}
+          options={['Subject1', 'Subject2', 'Subject3']} // TODO: replace with actual subjects from backend
```

Comment:
todoes in code is written in format `// @TODO ...`. Then it will your IDE can detect it as TODO item and display it in IDE tools